subprojects/snapshots/src/main/java/org/gradle/internal/change/DefaultFileChange.java

```diff
@@ -116,6 +120,19 @@ public boolean isRemoved() {
         return change == ChangeTypeInternal.REMOVED;
     }
 
+    public static org.gradle.work.FileType convertToPublicType(FileType fileType) {
+        switch (fileType) {
+            case RegularFile:
+                return org.gradle.work.FileType.FILE;
+            case Directory:
+                return org.gradle.work.FileType.DIRECTORY;
+            case Missing:
+                return org.gradle.work.FileType.MISSING;
+            default:
+                throw new AssertionError("Unsupported file type:" + fileType);
+        }
+    }
```

Comment:
This could be a `toPublicType()` method on `FileType`. But if the two types are practically the same, shouldn't we just use the public type internally, too?

Reply:
We can't, since our internal `FileType` lives in `baseServices`, while the public type lives in `coreApi`. I don't want to move the public type to `baseServices`.

Reply:
Why not?

Reply:
I could put the type in `org.gradle.api.file` in `baseServices`. I don't know yet if we need to evolve the internal and the public type separately. It felt like a good idea to keep the usage of the public type limited to the new incremental tasks API for now.

WDYT?

Reply:
Alright. It's easy to change this later if we decide to.

Reply:
I moved `FileType` to `org.gradle.api.file` and there is now a `toPublicType()` method on `org.gradle.internal.file.FileType`.